Testscripts/Windows/PERF-STORAGE-FIO-HB.ps1

```diff
@@ -0,0 +1,316 @@
+SetTestStateCompleted
+"@
+		Set-Content "$LogDir\fio2Command.sh" $fio2Command
+
+		#region Upload files to VM
+		foreach ($VMData in $AllVMData) {
+			Copy-RemoteFiles -uploadTo $VMData.PublicIP -port $VMData.SSHPort -files "$constantsFile,$($CurrentTestData.files),$LogDir\fio*.sh" -username $user -password $password -upload
+			Write-LogInfo "Copied the script files to the VM"
+		}
+		#endregion
+
+		# Configuration for the hibernation
+		Write-LogInfo "New kernel compiling..."
+		Run-LinuxCmd -ip $AllVMData.PublicIP -port $AllVMData.SSHPort -username $user -password $password -command "./SetupHbKernel.sh" -RunInBackground -runAsSudo -ignoreLinuxExitCode:$true | Out-Null
+		Write-LogInfo "Executed SetupHbKernel script inside VM"
+
+		# Wait for kernel compilation completion. 90 min timeout
+		$timeout = New-Timespan -Minutes $maxKernelCompileMin
+		$sw = [diagnostics.stopwatch]::StartNew()
+		while ($sw.elapsed -lt $timeout){
+			$vmCount = $AllVMData.Count
+			Wait-Time -seconds 30
+			$state = Run-LinuxCmd -ip $AllVMData.PublicIP -port $AllVMData.SSHPort -username $user -password $password "cat /home/$username/state.txt"
+			if ($state -eq "TestCompleted") {
+				$kernelCompileCompleted = Run-LinuxCmd -ip $AllVMData.PublicIP -port $AllVMData.SSHPort -username $user -password $password "cat ~/constants.sh | grep setup_completed=0"
+				if ($kernelCompileCompleted -ne "setup_completed=0") {
+					Write-LogErr "SetupHbKernel.sh finished on $($AllVMData.RoleName) but setup was not successful!"
+				} else {
+					Write-LogInfo "SetupHbKernel.sh finished on $($AllVMData.RoleName)"
+					$vmCount--
+				}
+				break
+			} elseif ($state -eq "TestSkipped") {
+				Write-LogInfo "SetupHbKernel.sh finished with SKIPPED state!"
+				$resultArr = $resultSkipped
+				$currentTestResult.TestResult = Get-FinalResultHeader -resultarr $resultArr
+				return $currentTestResult.TestResult
+			} elseif ($state -eq "TestFailed") {
+				Write-LogErr "SetupHbKernel.sh didn't finish successfully!"
+				$resultArr = $resultFail
+				$currentTestResult.TestResult = Get-FinalResultHeader -resultarr $resultArr
+				return $currentTestResult.TestResult
+			} elseif ($state -eq "TestAborted") {
+				Write-LogInfo "SetupHbKernel.sh finished with Aborted state!"
+				$resultArr = $resultAborted
+				$currentTestResult.TestResult = Get-FinalResultHeader -resultarr $resultArr
+				return $currentTestResult.TestResult
+			} else {
+				Write-LogInfo "SetupHbKernel.sh is still running in the VM!"
+			}
+		}
+		if ($vmCount -le 0){
+			Write-LogInfo "SetupHbKernel.sh is done"
+		} else {
+			throw "SetupHbKernel.sh didn't finish in the VM!"
+		}
+
+		# Reboot VM to apply swap setup changes
+		Write-LogInfo "Rebooting All VMs!"
+		$TestProvider.RestartAllDeployments($AllVMData)
+
+		# Check the VM status before hibernation
+		$vmStatus = Get-AzVM -Name $vmName -ResourceGroupName $rgName -Status
+		if ($vmStatus.Statuses[1].DisplayStatus = "VM running") {
```

Comment:
$vmStatus.Statuses[1].DisplayStatus -eq "VM running"
You need change all on = to -eq, = is for value assignment

Reply:
Fixed